examples/petsc/bpsraw.c

```diff
@@ -725,33 +796,52 @@ int main(int argc, char **argv) {
   user->op = op_apply;
   user->qdata = qdata;
   user->ceed = ceed;
+  user->memtype = memtyperequested;
+  if (memtyperequested == CEED_MEM_HOST) {
+    user->VecGetArray = VecGetArray;
+    user->VecGetArrayRead = VecGetArrayRead;
+    user->VecRestoreArray = VecRestoreArray;
+    user->VecRestoreArrayRead = VecRestoreArrayRead;
+  } else {
+    #ifdef PETSC_HAVE_CUDA
+    user->VecGetArray = VecCUDAGetArray;
+    user->VecGetArrayRead = VecCUDAGetArrayRead;
+    user->VecRestoreArray = VecCUDARestoreArray;
+    user->VecRestoreArrayRead = VecCUDARestoreArrayRead;
+    #endif
+  }
 
   ierr = MatCreateShell(comm, mnodes[0]*mnodes[1]*mnodes[2]*ncompu,
                         mnodes[0]*mnodes[1]*mnodes[2]*ncompu,
                         PETSC_DECIDE, PETSC_DECIDE, user, &mat); CHKERRQ(ierr);
+  if (user->memtype == CEED_MEM_DEVICE) {
+    #if PETSC_VERSION_GT(3,12,2)
+    ierr = MatShellSetVecType(mat, VECCUDA); CHKERRQ(ierr);
+    #endif
+  }
```

Comment:
I need to think on this. With `PETSC_VERSION_GT` automatically being true if PETSc isn't on a release, like when built from git, this is causing build errors for Azure. Travis is fine though.

Reply:
Your change probably won't go to maint so it'd be 3.13 before it is available. Simplest to wait on merging this until your PETSc MR merges.

Reply:
Fair enough. After the MR I suppose it'd be reasonable to assume that if PETSc wasn't a release, its the latest master.